### PR TITLE
Explicitly set etcd --logger to zap to match etcd 3.4+

### PR DESCRIPTION
**Pull request by tangcong**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:

etcd capnslog logger is deprecated in etcd 3.4 and removed in etcd 3.5.  features such as etcd 3.4 trace depend on zap logger.

### CODE REVIEW

File: cmd/kubeadm/app/phases/etcd/local.go [hunk 1]
```diff
@@ -223,6 +223,7 @@ func getEtcdCommand(cfg *kubeadmapi.ClusterConfiguration, endpoint *kubeadmapi.A
 		"peer-trusted-ca-file":        filepath.Join(cfg.CertificatesDir, kubeadmconstants.EtcdCACertName),
 		"peer-client-cert-auth":       "true",
 		"snapshot-count":              "10000",
+		"logger":                      "zap",
```
Comment (neolit123) [inline cmd/kubeadm/app/phases/etcd/local.go:226]: https://github.com/etcd-io/etcd/issues/11426
if the default logger changes in a future release, why do we have to preemtively switch to the new logger in our deployments of older etcd?

/hold

Comment (tangcong) [reply cmd/kubeadm/app/phases/etcd/local.go:226]: @jingyih does etcd 3.4 plan to set the default logger to zap? now, k8s already uses version 3.4,but logger is old, so trace feature cannot work,it plays a key role in trouble-shooting. if etcd 3.4 does not use zap, i think it is better if we explicitly set etcd --logger to zap.

Comment (jingyih) [reply cmd/kubeadm/app/phases/etcd/local.go:226]: etcd 3.4 supports both the capnslog (old) and zap (new) logger. I think it is user's choice to pick either (we recommend zap). If the logger is not explicitly set to zap here, is there a way for user to provide customized / additional flags to etcd?

Comment (neolit123) [reply cmd/kubeadm/app/phases/etcd/local.go:226]: IMO we should keep kubeadm's etcd.yaml static pod manifest without being explicit about the logger. currently kubeadm does support patching of the manifests using --experimental-kustomize, so the logger can be changed if the user really wants that.

Comment (tangcong) [reply cmd/kubeadm/app/phases/etcd/local.go:226]: thanks.i see.

Comment (tangcong) [reply cmd/kubeadm/app/phases/etcd/local.go:226]: zap is better than deprecated capnslog, it looks risk-free,why do not we set an optimal logger by default?  This is my pr motivation. thanks.

Comment (neolit123) [reply cmd/kubeadm/app/phases/etcd/local.go:226]: i agree that this is mostly risk free, but kubeadm should not be opinionated about the loggers in etcd. if the users want to change it they can modify their etcd.yaml.

**Pull request closed without merging**